modules/post/multi/recon/local_exploit_suggester.rb

```diff
@@ -74,28 +121,48 @@ def set_module_options(mod)
   end
 
   def is_module_wanted?(mod)
-    (
-      mod.kind_of?(Msf::Exploit::Local) &&
-      mod.has_check? &&
-      is_session_type_compat?(mod) &&
-      is_module_platform?(mod) &&
-      is_module_arch?(mod) &&
-      is_module_options_ready?(mod)
-    )
+    return false unless mod.kind_of?(Msf::Exploit::Local) && mod.has_check?
+
+    is_session_type_compat?(mod) &&
+    (@validate_platform ? is_module_platform?(mod) : true) &&
+    (@validate_arch ? is_module_arch?(mod) : true) &&
+    is_module_options_ready?(mod)
   end
 
   def setup
     return unless session
 
     print_status "Collecting local exploits for #{session.session_type}..."
 
+    # Let's cache these options so that we don't make excessive calls to the DB.
+    @validate_arch = datastore['ValidateArch']
+    @validate_platform = datastore['ValidatePlatform']
+    @validate_meterpreter_commands = datastore['ValidateMeterpreterCommands']
+    @print_tables = datastore['PrintTables']
```

Comment:
The comment here is confusing, `datastore` should not create calls to the database, the `datastore` is an in memory object.  This does reduce the calls to lookups in a hash.